packages/core/src/services/shellExecutionService.test.ts

```diff
@@ -356,6 +356,51 @@ describe('ShellExecutionService', () => {
       expect(result.aborted).toBe(true);
       // The process kill is mocked, so we just check that the flag is set.
     });
+
+    it('should send SIGTERM and then SIGKILL on abort', async () => {
+      vi.useFakeTimers();
+      const { result } = await simulateExecution(
+        'sleep 10',
```

Comment:
using a real sleep combined with fake timers seems a bit sketchy. can you comment on the intent of the sleep 10 or switch to using real timers for this test?